clients/venice-samza/src/main/java/com/linkedin/venice/samza/VeniceSystemProducer.java

```diff
@@ -156,8 +193,36 @@ public VeniceSystemProducer(
   }
 
   public VeniceSystemProducer(
-      String veniceD2ZKHost,
-      String d2ServiceName,
+      String primaryControllerColoD2ZKHost,
+      String primaryControllerD2ServiceName,
+      String storeName,
+      Version.PushType pushType,
+      String samzaJobId,
+      String runningFabric,
+      boolean verifyLatestProtocolPresent,
+      VeniceSystemFactory factory,
+      Optional<SSLFactory> sslFactory,
+      Optional<String> partitioners,
+      Time time) {
+    this(
+        primaryControllerColoD2ZKHost,
+        primaryControllerColoD2ZKHost,
+        primaryControllerD2ServiceName,
+        storeName,
+        pushType,
+        samzaJobId,
+        runningFabric,
+        verifyLatestProtocolPresent,
+        factory,
+        sslFactory,
+        partitioners,
+        time);
+  }
+
+  public VeniceSystemProducer(
```

Comment:
The overload constructors are for backward compatibility? Do we have any codes in the LI product which call the VeniceSystemProducer constructors directly?

Reply:
Yes. There are. Also, since they are not `private`, they are a part of the API, hence, we cannot delete them unless we upgrade the major versions (or minor versions in case of `0.x` releases).